### ISSUE TITLE
StackOverFlow when the target of a skeleton with a crossbow is killed

### ISSUE THREAD

**Issue by SrGnis** [The-Fireplace/Overlord#210]
**Describe the bug**
If a skeleton is pointing (arrow is loaded) a mob with a crossbow and the mob is killed before the skeleton can shot it, a overflow happens.

**Logs/Screenshots/Videos**
```
[18:01:37] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.class_148: Ticking entity
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:883) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:823) ~[client-intermediary.jar:?]
	at net.minecraft.class_1132.method_3748(class_1132.java:99) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.handler$zho000$modifiedRunLoop(MinecraftServer.java:2856) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:663) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:270) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.StackOverflowError
	at dev.the_fireplace.overlord.entity.ai.goal.combat.AbstractArmyCrossbowAttackGoal.method_6270(AbstractArmyCrossbowAttackGoal.java:58) ~[Overlord-3.1.2+1.18.1.jar:?]
	at dev.the_fireplace.overlord.entity.ai.goal.combat.AbstractArmyCrossbowAttackGoal.method_6270(AbstractArmyCrossbowAttackGoal.java:58) ~[Overlord-3.1.2+1.18.1.jar:?]
	at dev.the_fireplace.overlord.entity.ai.goal.combat.AbstractArmyCrossbowAttackGoal.method_6270(AbstractArmyCrossbowAttackGoal.java:58) ~[Overlord-3.1.2+1.18.1.jar:?]
	at dev.the_fireplace.overlord.entity.ai.goal.combat.AbstractArmyCrossbowAttackGoal.method_6270(AbstractArmyCrossbowAttackGoal.java:58) ~[Overlord-3.1.2+1.18.1.jar:?]
```

[Complete log](https://gist.github.com/SrGnis/54198dec23f1bdb68492bf800480f26a)

- Minecraft: 1.18.1
- Mod Version: 3.1.2
- Fireplace Lib Version: 6.2.2
- Fabric API: 0.44.0

**Comment by SrGnis** [comment]
The bug still happens using only Overlord and Fireplace lib.

**Comment by The-Fireplace** [comment]
Thanks for the details. Fixed in version 3.1.3
